### PR TITLE
completed array-reduce

### PR DESCRIPTION
**Pull request by m-Cho1**
image

### CODE REVIEW

File: array-reduce/main.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+console.log(`product: ${product}`);
+
+// balance:
+let initialValue = 0;
+const sumAllAmounts = (previous, current) => {
+  // console.log('previous: ', previous, ', current: ', current);
+  if (current.type === 'deposit') {
+    initialValue += current.amount;
+  } else if (current.type === 'withdrawal') {
+    initialValue -= current.amount;
```
Comment (scbowler) [inline array-reduce/main.js:41]: While this works it is kind of an oops it worked. Here you should be adding or subtracting from `previous`. then returning `previous`

**Review by scbowler — CHANGES_REQUESTED** [review]
Looking good, but an update to `sumAllAmounts` is needed. Let me know if you have any questions.

**Comment by m-Cho1** [comment]
Edited sumAllAmounts

**Review by scbowler — APPROVED** [review]
Great job implementing the feedback, looks good! 👍

**Comment by m-Cho1** [comment]
`// composite:
const combineProp = (previous, current) => {
  const result = Object.assign(previous, current);
  return result;
};
const composite = traits.reduce(combineProp, {});
console.log('composite: ', composite);
`

always add empty object literal if you are looking for to return new object

**Pull request merged**